fme/ace/test_ocean_train.py

```diff
@@ -0,0 +1,460 @@
+import dataclasses
+import re
+import tempfile
+
+import numpy as np
+import pytest
+import torch
+import xarray as xr
+
+from fme.ace.inference.evaluator import main as inference_evaluator_main
+from fme.ace.testing import DimSizes, MonthlyReferenceData, save_scalar_netcdf
+from fme.ace.testing.fv3gfs_data import get_nd_dataset
+from fme.ace.train.train import main as train_main
+from fme.core.coordinates import LatLonCoordinates
+from fme.core.testing.wandb import mock_wandb
+
+
+def save_ocean_nd_netcdf(
+    filename: str,
+    dim_sizes: DimSizes,
+    variable_names: list[str],
+    timestep_days: float = 1.0,
+    nz_levels: int = 2,  # Number of ocean levels for thetao_0, mask_0 etc.
+):
+    """
+    Saves a netCDF file with synthetic ocean data, including masks and NaNs.
+    Inspired by fme/coupled/data_loading/test_data_loader.py::_save_netcdf
+    and fme.ace.testing.fv3gfs_data.save_nd_netcdf.
```

Comment:
[nit] this path doesn't exist but it's just in the comment so I don't think it matters